api/api.yaml

```diff
@@ -3414,6 +3472,139 @@ paths:
+      tags: ["Domain Claiming"]
+      x-excluded: true
+      parameters:
+        - name: domain
+          in: query
+          required: true
+          schema:
+            type: string
+          description: Email address, domain name, or URL to check. The domain will be extracted and sanitized automatically.
+          example: xyz@withampersand.com
+      responses:
+        200:
+          description: Domain claim status
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ClaimedDomainResponse"
+        400:
+          description: Bad Request - Invalid input
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/InputValidationProblem"
+        404:
+          description: Domain not claimed
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        default:
+          description: Error
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+    post:
+      operationId: claimDomain
+      summary: Claim a domain
+      description: Claim a domain for an organization. Accepts email, domain, or URL.
+      tags: ["Domain Claiming"]
+      x-excluded: true
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - domain
+                - parentId
+                - parentType
+              properties:
+                domain:
+                  type: string
+                  description: Email address, domain name, or URL to claim. The domain will be extracted and sanitized automatically.
+                  example: xyz@withampersand.com
+                parentId:
+                  type: string
+                  description: ID of the parent entity claiming the domain
+                  example: orgId
+                parentType:
+                  type: string
+                  description: Type of the parent entity
+                  example: org
+        required: true
+      responses:
+        201:
+          description: Domain claimed successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ClaimedDomainResponse"
+        400:
+          description: Bad Request - Invalid input
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/InputValidationProblem"
+        409:
+          description: Domain already claimed
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
```

Comment:
Suppose they try to claim a domain that they themselves already claimed. Should that return a 409 or a 201? It's mainly a question about idempotent behavior.